htdocs/web_portal/controllers/downtime/add_downtime.php

```diff
@@ -71,53 +72,52 @@ function submit(\User $user = null) {
+        $params = [];
+
+        foreach ($downtimeInfo['SITE_LEVEL_DETAILS'] as $siteID) {
+            $downtimeInfo['Impacted_Services'] = $siteID['services'];
+            $downtimeInfo['Impacted_Endpoints'] = $siteID['endpoints'];
+
+            $params['submittedDowntimes'][$siteID['siteName']] =
+                $serv->addDowntime($downtimeInfo, $user);
+        }
 
-        $params['dt'] = $serv->addDowntime($downtimeInfo, $user);
         show_view("downtime/added_downtime.php", $params);
     }else{
         //Show user confirmation screen with their input
         $downtimeInfo = getDtDataFromWeb();
 
-        //Need to sort the impacted_ids into impacted services and impacted endpoints
-        $impactedids = $downtimeInfo['IMPACTED_IDS'];
+        list(
+            $siteLevelDetails,
+            $serviceWithEndpoints
+        ) = endpointToServiceMapping($downtimeInfo['IMPACTED_IDS']);
 
-        $services=array();
-        $endpoints=array();
+        // Delete the unsorted IDs from the downtime info
+        unset($downtimeInfo['IMPACTED_IDS']);
 
-        //For each impacted id sort between endpoints and services using the prepended letter
-        foreach($impactedids as $id){
-            if (strpos($id, 's') !== FALSE){
-                //This is a service id
-                $services[] = str_replace('s', '', $id); //trim off the identifying char before storing in array
-            }else{
-                //This is an endpoint id
-                $endpoints[] = str_replace('e', '', $id); //trim off the identifying char before storing in array
-            }
+        if (!count($siteLevelDetails) > 1) {
+            $downtimeInfo['SINGLE_TIMEZONE'] = true;
         }
 
-        unset($downtimeInfo['IMPACTED_IDS']); //Delete the unsorted Ids from the downtime info
-
-        $downtimeInfo['Impacted_Endpoints'] = $endpoints;
-
-
-        $serv = \Factory::getServiceService();
-
-        /** For endpoint put into downtime we want the parent service also. If a user has selected
-         * endpoints but not the parent service here we will add the service to maintain the link beteween
-         * a downtime having both the service and the endpoint.
-         */
-        foreach($downtimeInfo['Impacted_Endpoints'] as $endpointIds){
-           $endpoint = $serv->getEndpoint($endpointIds);
-           $services[] = $endpoint->getService()->getId();
-        }
-
-        //Remove any duplicate service ids and store the array of ids
-        $services = array_unique($services);
-
-        //Assign the impacted services and endpoints to their own arrays for us by the addDowntime method
-        $downtimeInfo['Impacted_Services'] = $services;
+        list(
+            $siteLevelDetails,
+            $serviceWithEndpoints
+        ) = addParentServiceForEndpoints(
+            $serviceWithEndpoints,
+            $siteLevelDetails,
+            false,
+            $downtimeInfo['DOWNTIME']
+        );
```

Comment:
can you look at different ways to split this assignment (and the other similar one) up over multiple lines? I read it as a single list initiation and missed that there was an assignment going on.